### PR TITLE
GH actions. build-and-test-all.yml: add make distdir as an additional step for build jobs

### PR DESCRIPTION
**Pull request by romeroalx**
### Short description
Adding `make distdir` as an additional step during the build of all components in `build-and-test-all.yml`

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: .github/workflows/build-and-test-all.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ jobs:
           restore-keys: auth-ccache-
       - run: inv ci-autoconf
       - run: inv ci-auth-configure
+      - run: inv ci-make-distdir
       - run: inv ci-auth-make-bear  # This runs under pdns/
```
Comment (Habbie) [inline .github/workflows/build-and-test-all.yml:54]: Can you run this build from inside the freshly generated distdir? That way we can detect missing files in dist.

**Review by Habbie — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test-all.yml (hunk 1 above)
Comment (Habbie) [reply .github/workflows/build-and-test-all.yml:54]: (which means running configure again, but not autoconf)

Comment (romeroalx) [reply .github/workflows/build-and-test-all.yml:55]: Done

Comment (romeroalx) [reply .github/workflows/build-and-test-all.yml:55]: The folder name comes from the BUILDER_VERSION variable that has been fixed in `tasks.py`:

```
def ci_autoconf(c):
    c.run('BUILDER_VERSION=0.0.0-git1 autoreconf -vfi')
```

If the goal is to configure and make from inside that directory, we might need to do it with `FUZZING_TARGETS=no` or do something in addition. I will have a look.

Thank you for the review!

**Review by Habbie — APPROVED** [review]

File: .github/workflows/build-and-test-all.yml [hunk 2]
```diff
@@ -50,13 +51,18 @@ jobs:
           restore-keys: auth-ccache-
       - run: inv ci-autoconf
       - run: inv ci-auth-configure
-      - run: inv ci-auth-make-bear  # This runs under pdns/
-      - run: ln -s .clang-tidy.full .clang-tidy
+      - run: inv ci-make-distdir
+      - run: inv ci-auth-configure
+        working-directory: pdns-${{ env.BUILDER_VERSION }}
```
Comment (rgacogne) [inline .github/workflows/build-and-test-all.yml:56]: Would it make sense to make that directory the default working directory for this job?

**Review by rgacogne — APPROVED** [review]
Looks good to me, nice work! I'm suggesting defining default directories but I'm fine with merging the PR as it is. It will make my life a bit harder with regard to generating code coverage but I'll deal with it :)

File: .github/workflows/build-and-test-all.yml (hunk 2 above)
Comment (romeroalx) [reply .github/workflows/build-and-test-all.yml]: Makes sense. I just updated the PR to include this change.

Thank you :)

**Review by rgacogne — APPROVED** [review]

**Pull request merged**